pkg/eventing/manager.go

```diff
@@ -204,6 +207,57 @@ func (em *EventingManager) GetBackendConfig() *env.BackendConfig {
 	return &em.backendConfig
 }
 
+func (em EventingManager) DeployPublisherProxyResources(
+	ctx context.Context,
+	eventing *v1alpha1.Eventing,
+	eppDeployment *appsv1.Deployment,
+	scheme *runtime.Scheme) error {
+	// define list of resources to create for EPP.
+	resources := []client.Object{
+		// ServiceAccount
+		newPublisherProxyServiceAccount(eppDeployment.Name, eppDeployment.Namespace, eppDeployment.Labels),
+		// ClusterRole
+		newPublisherProxyClusterRole(eppDeployment.Name, eppDeployment.Namespace, eppDeployment.Labels),
+		// ClusterRoleBinding
+		newPublisherProxyClusterRoleBinding(eppDeployment.Name, eppDeployment.Namespace, eppDeployment.Labels),
+		// Service to expose event publishing endpoint of EPP.
+		newPublisherProxyService(eppDeployment.Name, eppDeployment.Namespace, eppDeployment.Labels,
+			eppDeployment.Spec.Template.Labels),
+		// Service to expose metrics endpoint of EPP.
+		newPublisherProxyMetricsService(eppDeployment.Name, eppDeployment.Namespace, eppDeployment.Labels,
+			eppDeployment.Spec.Template.Labels),
+		// Service to expose health endpoint of EPP.
+		newPublisherProxyHealthService(eppDeployment.Name, eppDeployment.Namespace, eppDeployment.Labels,
+			eppDeployment.Spec.Template.Labels),
+	}
+
+	// add PeerAuthentication for istio.
+	paExists, err := em.kubeClient.PeerAuthenticationCRDExists(ctx)
+	if err != nil {
+		return err
+	}
+	if paExists {
+		resources = append(resources, newPublisherProxyPeerAuthentication(eppDeployment.Name, eppDeployment.Namespace,
+			eppDeployment.Labels,
+			eppDeployment.Spec.Template.Labels),
+		)
+	}
+
+	//// create the resources on k8s.
+	for _, object := range resources {
+		// add owner reference.
+		if err = controllerutil.SetControllerReference(eventing, object, scheme); err != nil {
+			return err
+		}
+
+		// patch apply the object.
+		if err = em.kubeClient.PatchApply(ctx, object); err != nil {
```

Comment:
We should not forget any field. We will test the upgrade from kyma to eventing-manager and compare if any field is missing. On the other hand if we intentionally want to remove some field, it will be easy to do so in this way.